src/Storages/System/StorageSystemZooKeeper.cpp

```diff
@@ -285,10 +288,12 @@ void StorageSystemZooKeeper::fillData(MutableColumns & res_columns, ContextPtr c
             }), nodes.end());
         }
 
-        std::vector<std::future<Coordination::GetResponse>> futures;
-        futures.reserve(nodes.size());
-        for (const String & node : nodes)
-            futures.push_back(zookeeper->asyncTryGet(path_part + '/' + node));
+        std::vector<std::future<Coordination::GetResponse>> futures(nodes.size());
```

Comment:
`reserve` and construtor accepting count are not the same thing.
Your implementation will defaulkt construct `nodes.size()` object while `reserve` will just reserve enough space for that amount of objects.
`reserve` should be returned.